### PR TITLE
Refactor Popover to MUI, Drop Beta Package, and Enhance Tests

### LINKED ISSUE #4917: Replace `base-ui-components/react/popover` beta package
**Is your feature request related to a problem? Please describe.**

We are using a beta version of the first major release of this package. It's not suitable for production code

**Describe the solution you'd like**

1. Consider using a non-beta version of an existing package suite
1. The `mui` suite that we are already using has a popup library that we should consider

**Describe alternatives you've considered**

- `mui` suite

### PR DESCRIPTION
**Pull request by rawadhossain**
**What kind of change does this PR introduce?**
Replace `base-ui-components/react/popover` beta package with `Popover` from `@mui/material`

**Issue Number:**
Fixes #4917 

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
- Replaced beta `Popover` from `@base-ui-components/react` with stable `Popover` from `@mui/material` in `Pledges.tsx` and `FundCampaignPledge.tsx`, updating state management and event handlers accordingly.
- Removed `@base-ui-components/react` dependency from `package.json` and updated test mocks to use `avatarURL` instead of `image` for consistency with component implementation.
- Improved test coverage by adding test cases for avatar rendering, data fallbacks, and popover interactions.

**Does this PR introduce a breaking change?**
No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

* **Chores**
  * Removed an unused UI dependency.

* **Improvements**
  * Popover interactions refined: anchor-based open/close behavior, Escape handling, and more reliable extra-users popup display.
  * Avatar handling improved: consistent avatar rendering, alt text, and robust fallbacks for main and extra users.

* **Tests**
  * Expanded test coverage for pledges and campaign pledges, covering avatars, popup interactions, missing data, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by palisadoes** [comment]
1. Please apply the code rabbit suggested changes, even the trivial ones. 
1. If they are either not applicable or suitable to maintaining or improving the features or maintainability of the app, then let me know.

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 1]
```diff
@@ -307,6 +320,8 @@ describe('Testing Campaign Pledge Screen', () => {
 
   afterEach(() => {
     vi.useRealTimers();
+
+    vi.clearAllMocks();
   });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:325]: Although global cleanup exists, this keeps local test isolation explicit and is harmless per your note.

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all resolved items. Validate whether any updated test files cover all use cases, and are valid. If the PR requires any changes, even trivial ones, then give your review a “requested changes” status. If no changes are required then give the PR an “approved” status.

**Comment by rawadhossain** [comment]
Noted. making the necessary changes and re-committing.

File: src/screens/UserPortal/Pledges/Pledge.spec.tsx [hunk 2]
```diff
@@ -1216,6 +1216,111 @@ describe('Testing User Pledge Screen', () => {
...
+
+    const popup = await screen.findByTestId('extra-users-popup');
+    expect(popup).toBeInTheDocument();
+    expect(popup.className).toContain('popupExtra');
+
+    await userEvent.keyboard('{Escape}');
+    await waitFor(() => {
+      expect(screen.queryByTestId('extra-users-popup')).not.toBeInTheDocument();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/UserPortal/Pledges/Pledge.spec.tsx:1323]: Addressed in commit c39ed6c

**Comment by rawadhossain** [comment]
@coderabbitai full review and reevaluate all resolved items. Validate whether any updated test files cover all use cases, and are valid. If the PR requires any changes, even trivial ones, then give your review a “requested changes” status. If no changes are required then give the PR an “approved” status.

**Review by palisadoes — APPROVED** [review]
